File: clients/java/client-impl/src/main/java/com/flipkart/grayskull/models/response/BatchGetSecretsResponse.java
package com.flipkart.grayskull.models.response;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

/**
 * Response body for the {@code POST /v1/secrets/batch} endpoint.
 */
@Getter
@AllArgsConstructor(onConstructor = @__(@JsonCreator))
@JsonIgnoreProperties(ignoreUnknown = true)
public final class BatchGetSecretsResponse {

    private final int updatedCount;
    private final List<UpdatedSecret> updatedSecrets;

    @Getter
    @AllArgsConstructor(onConstructor = @__(@JsonCreator))
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static final class UpdatedSecret {
        private final String projectId;
        private final String secretName;
        private final int dataVersion;
        private final String publicPart;
        private final String privatePart;
    }
}

File: clients/go/client-impl/internal/models/response/batch_get_secrets_response.go
package response

// BatchGetSecretsResponse is the response body for POST /v1/secrets/batch.
//
// Mirrors com.flipkart.grayskull.models.response.BatchGetSecretsResponse.
type BatchGetSecretsResponse struct {
	// UpdatedCount is the number of secrets in UpdatedSecrets. Provided by the
	// server for fast no-update detection without iterating the slice.
	UpdatedCount int `json:"updatedCount"`

	// UpdatedSecrets is the set of secrets whose dataVersion is strictly
	// greater than the client's lastKnownVersion. May be empty.
	UpdatedSecrets []UpdatedSecret `json:"updatedSecrets"`
}

// UpdatedSecret describes a single secret whose value the server believes the
// client should refresh. Mirrors
// com.flipkart.grayskull.models.response.BatchGetSecretsResponse.UpdatedSecret.
type UpdatedSecret struct {
	// ProjectID is the project that owns the secret.
	ProjectID string `json:"projectId"`

	// SecretName is the name of the secret within the project.
	SecretName string `json:"secretName"`

	// DataVersion is the new version of the secret. Always strictly greater
	// than the LastKnownVersion the client sent.
	DataVersion int32 `json:"dataVersion"`

	// PublicPart is the non-sensitive part of the secret payload.
	PublicPart string `json:"publicPart"`

	// PrivatePart is the sensitive part of the secret payload.
	PrivatePart string `json:"privatePart"`
}
